lib/DocumentManager.php

```diff
@@ -25,6 +26,12 @@
  *
  * This class is therefore an event dispatcher and a service container and has
  * no other business logic within it.
+ *
+ * All primary methods are wrapped in a try catch block, the exception is then
+ * wrapped in a `DocumentManagerException` with a reference to this document
+ * managers name.
+ *
+ * TODO: Create a proxy document manager which automatically wraps all calls.
```

Comment:
Rather than wrap the methods we could create a proxy document manager (`ExceptionCatchingDocumentManager`) which wraps *all* calls with th exception handling logic.

note that this is better in that we reduce duplication, increase test coverage (we only need to test one case) but increase magic by using `__call`